### PR TITLE
feat(compiler): add the duration division operator

### PR DESCRIPTION
**Pull request by EmileRolley**
> Related issue : #98 

#### Changelog

* Add the `/^` operator, including:
  * the new token `DIVDURATION`
  * the new runtime exception `IndivisableDurations`
  * new test files
* Fix the syntax cheat sheet for the english syntax.

### CODE REVIEW

File: src/catala/dcalc/interpreter.ml [hunk 1]
```diff
@@ -69,7 +69,7 @@ let rec evaluate_operator (ctx : Ast.decl_ctx) (op : A.operator Pos.marked)
           Errors.raise_multispanned_error "division by zero at runtime"
             [
               (Some "The division operator:", Pos.get_position op);
-              (Some "The null denominator:", Pos.get_position (List.nth args 2));
+              (Some "The null denominator:", Pos.get_position (List.nth args 1));
```
Comment (denismerigoux) [inline src/catala/dcalc/interpreter.ml:72]: This should be tweaked since #126 has been merged

File: src/catala/dcalc/interpreter.ml [hunk 2]
```diff
@@ -93,6 +93,21 @@ let rec evaluate_operator (ctx : Ast.decl_ctx) (op : A.operator Pos.marked)
...
     | A.Binop (A.Add KDate), [ ELit (LDate i1); ELit (LDuration i2) ] ->
         A.ELit (LDate Runtime.(i1 +@ i2))
+    | A.Binop (A.Div KDuration), [ ELit (LDuration i1); ELit (LDuration i2) ] -> (
+        try A.ELit (LRat Runtime.(i1 /^ i2)) with
+        | Division_by_zero ->
+            Errors.raise_multispanned_error "division by zero at runtime"
+              [
+                (Some "The division operator:", Pos.get_position op);
+                (Some "The null denominator:", Pos.get_position (List.nth args 1));
+              ]
```
Comment (denismerigoux) [inline src/catala/dcalc/interpreter.ml:103]: Same, use combinators from #126 ?

**Review by denismerigoux — APPROVED** [review]
Thanks again @EmileRolley ! You might want to revisit some of the code with #126 merged but otherwise perfect.

**Comment by EmileRolley** [comment]
> Thanks again @EmileRolley ! You might want to revisit some of the code with #126 merged but otherwise perfect.

You are welcome! Yes, I have it in mind and I'm updating it.

**Review by denismerigoux — APPROVED** [review]

**Pull request merged**